merge_sort.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import unicode_literals
+from time import time
+
+
+def merge_sort(init_list):
+    if len(init_list) <= 1:
+        return init_list
+    list_left = init_list[:len(init_list) / 2]
+    list_right = init_list[len(init_list) / 2:]
+    left = merge_sort(list_left)
+    right = merge_sort(list_right)
+    return merge(left, right)
+
+
+def merge(left, right):
+    return_list = []
+    while left and right:
+        if left[0] <= right[0]:
+            return_list.append(left[0])
+            left = left[1:]
```

Comment:
I don't know if making a slice of the list if cheaper than `.pop(0)`.  Looking at [this wiki page](https://wiki.python.org/moin/TimeComplexity) I kind of suspect it isn't.